src/yaml/aslYamlLanguageService.ts

```diff
@@ -181,11 +137,35 @@ function isChildOfStates(document: TextDocument, offset: number) {
 
         const positionForDoComplete = {...tempPositionForCompletions} // Copy position to new object since doComplete modifies the position
         const yamlCompletions = await completer.doComplete(processedDocument, positionForDoComplete, false)
+        // yaml-language-server does not output correct completions for retry/catch
+        // we need to overwrite the text
+        yamlCompletions.items.forEach(item => {
+            if (item.label === 'Catch') {
```

Comment:
Could be a bit DRYer e.g.

```js
ERROR_COMPLETION = { Catch: CATCH_INSERT, Retry: RETRY_INSERT }
yamlCompletions.items.forEach(item => { 
    if (Object.hasOwnProperty(item.label)) {
        item.insertText = ERROR_COMPLETION[item.label]
        if (item.textEdit) {
           item.textEdit.newText = ERROR_COMPLETION[item.label]
        }
    }
})
```

Reply:
So the code above would cause bugs as you are overwritting all completions not just Retry/Catch. The ones that are not Retry/Catch would be overwritten with  `undefinded`.

Reply:
I made it a bit dry'er by abstracting away overwriting logic to a separate function.